### ISSUE TITLE
Changes in forecast horizon for future data changes test accuracy

### ISSUE THREAD

**Issue by Pushkaran-P** [ourownstory/neural_prophet#1329]
Hi team,

Does changes in forecast horizon affect the predicted values in my test set in my code ? Normally this should'nt make sense.
Neuralprophet Version: 0.6.0

Dataframe : 
[df.csv](https://github.com/ourownstory/neural_prophet/files/11485841/df.csv)

Code for replication :
```
import pandas as pd
import numpy as np
import time
import math

from neuralprophet import NeuralProphet, set_log_level, save, load, utils

def mape(actual, predicted):
    actual, predicted = np.array(actual), np.array(predicted)
    mape = np.mean(np.abs((actual - predicted) / actual))
    return mape

df = pd.read_csv('df.csv',index_col=0)
test_horizon = 3

df_train = df[:-test_horizon]
forecasts = test_horizon
n_changepoints = math.floor(len(df_train)/12)
m = NeuralProphet(weekly_seasonality=False,daily_seasonality=False, yearly_seasonality="auto", growth = "linear",
            n_changepoints=n_changepoints, global_normalization=True, n_lags=forecasts*2, 
            trend_reg=0, 
            seasonality_reg = 0,
            normalize = "standardize",
            #ar_layers = [forecasts],
            n_forecasts=forecasts
            )
utils.set_random_seed(seed=0)
fit_model = m.fit(df_train, freq="MS", minimal = True, progress = None)

past = m.predict(df_train)
pred_train = past.yhat1[-15:]

y_train = df['y'][-15:]
train_mape = mape(y_train, pred_train)
train_acc = 1-train_mape

future = m.make_future_dataframe(df_train, periods=forecasts)
forecast = m.predict(future)
df_fc = m.get_latest_forecast(forecast)
pred_test = df_fc['origin-0'][:test_horizon]
print(pred_test)

y_test = df['y'][-test_horizon:]
test_mape = mape(y_test, pred_test)
test_acc = 1-test_mape
print(test_acc)
```
I get a test accuracy of 48%
However when i only increase the n_forecasts parameter ( n_forecasts = forecasts + 9 ) my accuracy decreases a lot
```
m = NeuralProphet(weekly_seasonality=False,daily_seasonality=False, yearly_seasonality="auto", growth = "linear",
            n_changepoints=n_changepoints, global_normalization=True, n_lags=forecasts*2, 
            trend_reg=0, 
            seasonality_reg = 0,
            normalize = "standardize",
            #ar_layers = [forecasts],
            n_forecasts=forecasts + 9
            )
```
Doing the following doesnt seem to help either
```
forecasts = test_horizon + 9
n_changepoints = math.floor(len(df_train)/12)
m = NeuralProphet(weekly_seasonality=False,daily_seasonality=False, yearly_seasonality="auto", growth = "linear",
            n_changepoints=n_changepoints, global_normalization=True, n_lags=forecasts*2, 
            trend_reg=0, 
            seasonality_reg = 0,
            normalize = "standardize",
            #ar_layers = [forecasts],
            n_forecasts=forecasts
            )
```
Why is this happening and also what can i do to improve the forecasts of my model. Thanks in advance

**Comment by ourownstory** [comment]
Hi @Pushkaran-P 
Changing your forecast horizon is expected to impact your forecasting accuracy. It is generally harder to predict the more distant than near future. 

I noticed two places in your code that may need your attention:
- You seem to be setting a large amount of change-points for the trend. That may lead the model to overfit the trend to a point where the trend just follows the data. I suggest keeping your changepoints to a range of 0-20 unless you carefully tune it. 
- your train accuracy calculation seems to have misplaced indices of predicted vs ground truth.
